### PR TITLE
You can now FIX your MALF AI. 

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

This PR makes it so that you can fix your malfunctional (naughty) AI instead of wiping and RR'ing them.
Yes, to get to this point you need to defeat the AI. Keep the player in the round, everyone benefits, etc.
https://cdn.discordapp.com/attachments/610319137012580383/1049887877972099083/dreamseeker_QRntph4a7Z.mp4
## How This Contributes To The Skyrat Roleplay Experience

Lets players do more without the fear of sitting in ghost and getting jaded so fast by the inevitable "malf bad" gang busting your doors down
SftFpIRyQm

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  

https://user-images.githubusercontent.com/77420409/206078823-826e331b-8242-49e5-abfa-ad774ea8fd75.mp4

https://user-images.githubusercontent.com/77420409/206079197-d8e38bed-609c-467f-b6e0-4776f55be4c2.mp4

</details>

## Changelog

🆑 
add: You can now un-malf your malf-AI!
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/AiRestorer.tsx [hunk 1]
```diff
@@ -84,6 +84,14 @@ export const AiRestorerContent = (props, context) => {
             mt={1}
             onClick={() => act('PRG_beginReconstruction')}
           />
+          <Button
```
Comment (RimiNosha) [inline tgui/packages/tgui/interfaces/AiRestorer.tsx:88]: `{/* SKYRAT EDIT ADDITION - Malf Remover */}`

Above this line, and the same with an end comment after the button, just to make it easier to figure out if this is an us thing in the future.

**Comment by YakumoChen** [comment]
Round removing antags is bad, but I think de-antagging them is much worse. I rarely see crew actually fully keep a malf AI Round Removed. It removes all of the intrigue of the malf AI trying to plot a way to get revenge and get back in its seat of power, and really destroys any sort of story you can have with that sort of thing.

I'd rather leave de-antagging a thing only within the realm of admins, not players, so that the antag can actually consent to being unantagged, instead of just being a button the RD presses to make funny AI boring again.

Having it be an "antivirus" doesn't make much sense by itself either.

**Comment by nevimer** [comment]
> 



> Round removing antags is bad, but I think de-antagging them is much worse. I rarely see crew actually fully keep a malf AI Round Removed. It removes all of the intrigue of the malf AI trying to plot a way to get revenge and get back in its seat of power, and really destroys any sort of story you can have with that sort of thing.
> 
> I'd rather leave de-antagging a thing only within the realm of admins, not players, so that the antag can actually consent to being unantagged, instead of just being a button the RD presses to make funny AI boring again.
> 
> Having it be an "antivirus" doesn't make much sense by itself either.

I mean, if you read the PR, the AI has to consent.

**Comment by YakumoChen** [comment]
I noticed the caveat, was going to edit my post.

I still don't like it. Having this just encourages the crew to force the choice of "De-antag or die", instead of having to either live with the morality of RRing a player or holding them alive with the threat of an unexpected return.

File: tgui/packages/tgui/interfaces/AiRestorer.tsx (hunk 1 above)
Comment (nevimer) [reply tgui/packages/tgui/interfaces/AiRestorer.tsx:88]: done thank u

**Comment by nevimer** [comment]
> I noticed the caveat, was going to edit my post.
> 
> I still don't like it. Having this just encourages the crew to force the choice of "De-antag or die", instead of having to either live with the morality of RRing a player or holding them alive with the threat of an unexpected return.

My experience has been being wiped or stuck in a card for good, boringly, for the rest of the round. Nobody's giving antags any mercy these days, as a trend.

**Comment by YakumoChen** [comment]
You can still do a decent amount from a card, and that doesn't stop you from finding a potential ally to get their hands on you.

**Comment by nevimer** [comment]
> 



> You can still do a decent amount from a card, and that doesn't stop you from finding a potential ally to get their hands on you.

Cards by default stop you from doing anything except talking locally and on binary. You can't use malf abilities in cards either.

**Comment by RatFromTheJungle** [comment]
👍 good change, less insistence on RR during longer-rounds if they get snagged early

**Comment by YakumoChen** [comment]
> > 
> 
> > You can still do a decent amount from a card, and that doesn't stop you from finding a potential ally to get their hands on you.
> 
> Cards by default stop you from doing anything except talking locally and on binary. You can't use malf abilities in cards either.

last time I played malf (and it has been awhile, I admit) you could do malf abilities on a card just fine.

**Pull request closed without merging**